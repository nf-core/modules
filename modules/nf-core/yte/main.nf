process YTE {
    tag "$meta.id"
    label 'process_single'

    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/yte:1.5.4--2466971526dcd2a8':
        'community.wave.seqera.io/library/yte:1.5.4--36d16ca4bab836c1' }"

    input:
    tuple val(meta), path(template)

    output:
    tuple val(meta), path("*.yaml"), emit: rendered
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    VERSION = "1.5.4" // WARN: Version information not provided by tool on CLI. Please update this string when bumping
    """
    yte < ${template} > ${prefix}.yaml

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        yte: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    VERSION = "1.5.4" // WARN: Version information not provided by tool on CLI. Please update this string when bumping
    """
    touch ${prefix}.yaml

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        yte: $VERSION
    END_VERSIONS
    """
}
