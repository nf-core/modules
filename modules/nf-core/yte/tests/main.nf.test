nextflow_process {

    name "Test Process YTE"
    script "../main.nf"
    process "YTE"

    tag "modules"
    tag "modules_nfcore"
    tag "yte"

    test("yaml - map") {

        when {
            process {
                """
                input[0] = [
                    [ id:'template' ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/1d7285ed7dea21f2d4079f7daa196b9f17bc6e85/data/generic/config/config_template.yte.yaml', checkIfExists: true)
                ]
                input[1] = []
                input[2] = [ 'sex': 'female' ] // map
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("yaml - map - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'template' ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/1d7285ed7dea21f2d4079f7daa196b9f17bc6e85/data/generic/config/config_template.yte.yaml', checkIfExists: true)
                ]
                input[1] = []
                input[2] = [ 'sex': 'female' ] // map
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                    ).match() }
            )
        }

    }

    test("yaml - map_file") {

        when {
            process {
                """
                input[0] = [
                    [ id:'template' ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/1d7285ed7dea21f2d4079f7daa196b9f17bc6e85/data/generic/config/config_template.yte.yaml', checkIfExists: true)
                ]
                input[1] = [file ('https://raw.githubusercontent.com/nf-core/test-datasets/811dd3fa80dbac64b14f07413ad78bb2bebe0323/data/generic/config/yte_map_file.yaml', checkIfExists: true)]
                input[2] = [] // map
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("yaml - map_file - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'template' ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/1d7285ed7dea21f2d4079f7daa196b9f17bc6e85/data/generic/config/config_template.yte.yaml', checkIfExists: true)
                ]
                input[1] = [file ('https://raw.githubusercontent.com/nf-core/test-datasets/811dd3fa80dbac64b14f07413ad78bb2bebe0323/data/generic/config/yte_map_file.yaml', checkIfExists: true)]
                input[2] = [] // map
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
