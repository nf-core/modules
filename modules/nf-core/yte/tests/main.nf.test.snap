{
    "yaml - map_file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d85fd82470de636a7d97b869d6c32fbc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ],
                "rendered": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d85fd82470de636a7d97b869d6c32fbc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-14T11:19:26.872521"
    },
    "yaml - map": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d85fd82470de636a7d97b869d6c32fbc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ],
                "rendered": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d85fd82470de636a7d97b869d6c32fbc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-14T11:19:16.956644"
    },
    "yaml - map_file - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ],
                "rendered": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-14T11:19:31.39573"
    },
    "yaml - map - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ],
                "rendered": [
                    [
                        {
                            "id": "template"
                        },
                        "template.yaml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,38d0aa4f7c9871ffc21a908df10e3b9d"
                ]
            },
            {
                "YTE": {
                    "yte": "1.9.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-14T11:20:56.570353"
    }
}